components/button/button-add.js

```diff
@@ -15,6 +15,10 @@ import { PropertyRequiredMixin } from '../../mixins/property-required/property-r
 class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(LitElement))) {
 	static get properties() {
 		return {
+			/**
+			 * ONLY used when text-visible is FALSE. When true, icon is only visible when component receives hover or focus. When false (default), icon is always visible.
+			 */
+			iconOnlyVisibleOnHoverFocus: { type: Boolean, reflect: true, attribute: 'icon-only-visible-on-hover-focus' },
```

Comment:
I've removed the 2 new variables for the always/never/hover-focus states and replaced with a bool since there are now only the 2 possible states for icon.

Reply:
Hmm... would a single tri-state attribute make more sense? Something like `mode` which has the possible values of: `icon` (default), `icon-and-text` (equivalent to the current `text-visible`) or `icon-when-interacted`? You could still reflect it to target the different states with CSS attribute selectors.

It's always a bit of a code smell to me when we have attributes that only apply under certain circumstances and if ignored can cause the thing to be in a weird state.

Reply:
Good call, I like that better. It seems more clear.